### ISSUE TITLE
Joining :all rooms not working

### ISSUE THREAD

**Issue by amaltson** [jimmycuadra/lita-hipchat#19]
I'm using Lita 4.0.2 and the Lita HipChat Handler 2.0.1. I'm able to get Lita to join specific rooms, but for some reason, when I get Lita to join `:all`, it crashes with the stacktrace below. Thanks in advance for any help!

```
app[stderr.0]: D, [2014-11-19T18:48:37.712231 #131] DEBUG -- : SENDING:
app[stderr.0]:     <presence xmlns='jabber:client'><show>chat</show></presence>
app[stderr.0]: D, [2014-11-19T18:48:37.713097 #131] DEBUG -- : SENDING:
app[stderr.0]:     <iq from='1_3@chat.btf.hipchat.com/bot' id='3918' to='btf.hipchat.com' type='get' xmlns='jabber:client'><query xmlns='http://jabber.org/protocol/disco#items'/></iq>
app[stderr.0]: D, [2014-11-19T18:48:37.744950 #131] DEBUG -- : RECEIVED:
app[stderr.0]:     <presence from='1_3@chat.btf.hipchat.com/bot' to='1_3@chat.btf.hipchat.com'><show>chat</show></presence>
app[stderr.0]: D, [2014-11-19T18:48:37.745549 #131] DEBUG -- : PROCESSING:
app[stderr.0]:     <presence from='1_3@chat.btf.hipchat.com/bot' to='1_3@chat.btf.hipchat.com' xmlns='jabber:client'><show>chat</show></presence> (Jabber::Presence)
app[stderr.0]: D, [2014-11-19T18:48:37.745613 #131] DEBUG -- : TRYING stanzacbs...
app[stderr.0]: D, [2014-11-19T18:48:37.745653 #131] DEBUG -- : TRYING message/iq/presence/cbs...
app[stderr.0]: D, [2014-11-19T18:48:37.747372 #131] DEBUG -- : RECEIVED:
app[stderr.0]:     <iq from='btf.hipchat.com' id='3918' to='1_3@chat.btf.hipchat.com/bot' type='error'><query xmlns='http://jabber.org/protocol/disco#items'/><error code='400' type='modify'><bad-request xmlns='urn:ietf:params:xml:ns:xmpp-stanzas'/></error></iq>
app[stderr.0]: /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/stream.rb:450:in `send_with_id': bad-request:  (Jabber::ServerError)
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/muc/helper/mucbrowser.rb:82:in `muc_rooms'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/bundler/gems/lita-hipchat-6ad1f2d5dc95/lib/lita/adapters/hipchat/connector.rb:60:in `list_rooms'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/bundler/gems/lita-hipchat-6ad1f2d5dc95/lib/lita/adapters/hipchat.rb:72:in `rooms'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/bundler/gems/lita-hipchat-6ad1f2d5dc95/lib/lita/adapters/hipchat.rb:44:in `run'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-4.0.2/lib/lita/robot.rb:69:in `run'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-4.0.2/lib/lita.rb:75:in `run'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-4.0.2/lib/lita/cli.rb:77:in `start'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-4.0.2/bin/lita:6:in `<top (required)>'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/bin/lita:23:in `load'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/bin/lita:23:in `<main>'
```

**Comment by jimmycuadra** [comment]
Did you have any luck with this?

**Comment by amaltson** [comment]
@jimmycuadra sorry, completely forgot to check.

I recently upgraded us to HipChat Server 1.1.5. Also running Lita 4.1.0 and lita-hipchat 2.1.1. Unfortunately `:all` is still not working, but specifying the rooms works fine. Stacktrace:

```
app[stderr.0]: D, [2015-01-16T05:10:22.781870 #130] DEBUG -- : RECEIVED:
app[stderr.0]:     <iq from='btf.hipchat.com' id='3419' to='1_3@chat.btf.hipchat.com/bot||proxy|devchat.ms.otpp.net|5222' type='error'><query xmlns='http://jabber.org/protocol/disco#items'/><error code='400' type='modify'><bad-request xmlns='urn:ietf:params:xml:ns:xmpp-stanzas'/></error></iq>
app[stderr.0]: /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/stream.rb:450:in `send_with_id': bad-request:  (Jabber::ServerError)
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/muc/helper/mucbrowser.rb:82:in `muc_rooms'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-hipchat-2.1.1/lib/lita/adapters/hipchat/connector.rb:60:in `list_rooms'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-hipchat-2.1.1/lib/lita/adapters/hipchat.rb:72:in `rooms'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-hipchat-2.1.1/lib/lita/adapters/hipchat.rb:44:in `run'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-4.1.0/lib/lita/robot.rb:69:in `run'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-4.1.0/lib/lita.rb:79:in `run'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-4.1.0/lib/lita/cli.rb:77:in `start'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/gems/lita-4.1.0/bin/lita:6:in `<top (required)>'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/bin/lita:23:in `load'
app[stderr.0]: 	from /home/stackato/app/vendor/bundle/ruby/2.1.0/bin/lita:23:in `<main>'
stackato[dea_ng.0]: Unable to successfully start instance (internal error (fence job failed))
```

**Comment by amaltson** [comment]
Is there a HipChat API I can poke directly to see if it's working?

**Comment by jimmycuadra** [comment]
No, HipChat doesn't have its own real time API, so lita-hipchat is implemented using XMPP. It'd be nice if Atlassian provided a public instance of HipChat Server to test against. I found it pretty difficult to set up one of my own, even with the images they provide.

**Comment by amaltson** [comment]
@jimmycuadra hmm, I haven't had any issues setting it up, but then again I have a workstation with ESXi installed. It should run in VMware Fusion though...

**Comment by amaltson** [comment]
@jimmycuadra is there anything I can do on my end to debug this further?

**Comment by jimmycuadra** [comment]
Sorry I haven't been able to spend more time on this! What I would do next, assuming I had an environment where this was happening set up, I'd clone lita-hipchat and deploy Lita with the Gemfile pointing at the local path for lita-hipchat, then start inserting `binding.pry` statements in the code to poke around and figure out what's going on. Clearly the lita-hipchat code is making some assumption about how the server will respond that isn't always true, but it's hard to know what that might be without playing around a bit. XMPP is kinda hairy, and the xmpp4r gem doesn't help much. :P

**Comment by jimmycuadra** [comment]
I never did figure this out, but the `rooms` configuration attribute is now deprecated in the just-released lita-hipchat 3.0: https://github.com/jimmycuadra/lita-hipchat/releases/tag/v3.0.0.

You can use the runtime commands for room management and this should no longer cause problems for you.
